pkg/injector/pod_patch.go

```diff
@@ -776,6 +775,19 @@ func getSidecarContainer(annotations map[string]string, id, daprSidecarImage, im
 
 	c.Env = append(c.Env, utils.ParseEnvString(annotations[daprEnvKey])...)
 
+	// This is a special case that requires administrator privileges in Windows containers
+	// to install the certificates to the root store. If this environment variable is set,
+	// the container security context should be set to run as administrator.
+	for _, env := range c.Env {
+		if env.Name == "SSL_CERT_DIR" {
+			userName := "ContainerAdministrator"
+			c.SecurityContext.WindowsOptions = &corev1.WindowsSecurityContextOptions{
```

Comment:
why is the windows security option also added for linux containers if SSL CERT DIR is given?
Also here only an implicit consent is used to elevate container user to Admin ie SSL CERT DIR is given ... shouldn't there be an explicit consent given by the user say set an annotation so that user explicitly turns on this feature?

Reply:
> why is the windows security option also added for linux containers if SSL CERT DIR is given?

We have no way to know if a container is Linux- or Windows- based

> shouldn't there be an explicit consent given by the user say set an annotation so that user explicitly turns on this feature?

IMHO, using ContainerAdministrator is necessary and sufficient for, and only for, using custom TLS certs. Perhaps this should be explained in the docs: if you enable that feature (and only in that case), the user within the container will be administrator.

Reply:
I agree that we need docs for this.

Reply:
https://github.com/dapr/docs/issues/2570 docs issue.